src/gtk/textctrl.cpp

```diff
@@ -2153,13 +2191,24 @@ wxSize wxTextCtrl::DoGetSizeFromTextSize(int xlen, int ylen) const
     //multiline
     else
     {
+        // add space for vertical scrollbar
+        if ( m_scrollBar[1] && !(m_windowStyle & wxTE_NO_VSCROLL) )
+            tsize.IncBy(GTKGetPreferredSize(GTK_WIDGET(m_scrollBar[1])).x + 3, 0);
+
         // height
         if ( ylen <= 0 )
+        {
             tsize.y = 1 + cHeight * wxMax(wxMin(GetNumberOfLines(), 10), 2);
+            // add space for horizontal scrollbar
+            if ( m_scrollBar[0] && (m_windowStyle & wxHSCROLL) )
+                tsize.IncBy(0, GTKGetPreferredSize(GTK_WIDGET(m_scrollBar[0])).y + 3);
+        }
 
-        GtkRequisition req;
-        gtk_widget_get_preferred_size(m_widget, &req, nullptr);
-        tsize.IncTo(wxSize(req.width, req.height));
+        if ( !HasFlag(wxBORDER_NONE) )
+        {
+            // hardcode borders, margins, etc
+            tsize.IncBy(5, 4);
```

Comment:
Where do these numbers come from?

Reply:
@MaartenBent ,
Same as with previous comment

Reply:
It is new code, added by you. So please explain why these values.